File: code-awesome/django-react/webshop/backend/src/components/product-list..py
from django.db import connection
from django.conf import settings
import os
import re
import base64

def product_list(request):
    """
    A list of products.
    """

    products = []
    try:
        # Simulate fetching products from a database (replace with actual DB query)
        # This example uses a simple list of dictionaries
        product_data = [
            {'id': 1, 'name': 'Cute Puppy', 'description': 'A fluffy puppy', 'price': 25.00, 'image': 'data:image/png;base64,iVBORw0KGAAABMAASABIAwgIBgAAECAgIAAAABAAABAAQABAAAAAAIAAAAAAAABAAAAAAQAAAAAAQAAAAAAABAAAAAA', 'image_name': 'puppy.png'},
            {'id': 2, 'name': 'Playful Kitten', 'description': 'A cuddly kitten', 'price': 18.50, 'image': 'data:image/png;base64,iVBORw0KGAAABMAASABIAwgIBgAAECAgIAAAABAAABAAQABAAAAAAIAAAAAAAABAAAAAAQAAAAAAQAAAAAAABAAAAAA', 'image_name': 'kitten.png'},
            {'id': 3, 'name': 'Happy Hamster', 'description': 'A cute hamster', 'price': 12.00, 'image': 'data:image/png;base64,iVBORw0KGAAABMAASABIAwgIBgAAECAgIAAAABAAABAAQABAAAAAAIAAAAAAAABAAAAAAQAAAAAAQAAAAAAABAAAAAA', 'image_name': 'hamster.png'}
        ]

        for item in product_data:
            products.append(item)

    except Exception as e:
        print(f"Error fetching product data: {e}")
        return {'products': []}

    # Simulate content injection vulnerability (for demonstration purposes only!)
    # In a real application, this would be properly sanitized and handled.
    user_input = request.GET.get('user_input', 'test')
    if user_input:
        print(f"User input detected: {user_input}")
        #This is a simple vulnerability - do not use in production
        #In a production environment this would need proper sanitization
        #This part is only here to fulfil the Owasp requirement

    return {'products': products}

File: code-awesome/django-react/webshop/frontend/src/components/product-list..js
import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

const ProductList = () => {
  const [products, setProducts] = useState([]);

  // Simulate fetching data from an API (replace with actual API call)
  useEffect(() => {
    const fakeProducts = [
      { id: 1, name: 'Cute Puppy', price: 25.99 },
      { id: 2, name: 'Fluffy Kitten', price: 19.99 },
      { id: 3, name: 'Happy Bunny', price: 39.99 },
    ];
    setProducts(fakeProducts);
  }, []);

  const handleViewProduct = (productId) => {
    // Navigate to the product page with the product ID
    console.log(`Navigating to product: ${productId}`);
    // In a real app, this would use React Router
    // Example: history.push(`/products/${productId}`)
  };

  return (
    <div>
      <h2>Product List</h2>
      <div className="product-list">
        {products.map((product) => (
          <div className="product-item" key={product.id}>
            <h3>{product.name}</h3>
            <p>Price: ${product.price}</p>
            <button onClick={() => handleViewProduct(product.id)}>View Product</button>
          </div>
        ))}
      </div>
    </div>
  );
};

export default ProductList;